sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -840,8 +840,8 @@ Mono<Response<Void>> deleteEntityWithResponse(String partitionKey, String rowKey
         context = TableUtils.setContext(context);
         eTag = ifUnchanged ? eTag : "*";
 
-        if (isNullOrEmpty(partitionKey) || isNullOrEmpty(rowKey)) {
-            return monoError(logger, new IllegalArgumentException("'partitionKey' and 'rowKey' cannot be null."));
+        if (partitionKey == null || rowKey == null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'partitionKey' and 'rowKey' cannot be null"));
```

Comment:
Add `@throws IllegalArgumentException` to the javadocs in both sync and async clients.

Reply:
Any particular reason we're moving away from using `monoError` like in the rest of the operations in the client?

Reply:
That's not supposed to be there, it's supposed to be monoError